core/src/main/java/io/fabric8/maven/core/config/MappingConfig.java

```diff
@@ -0,0 +1,24 @@
+package io.fabric8.maven.core.config;
+
+import org.apache.maven.plugins.annotations.Parameter;
+
+public class MappingConfig {
+
+    @Parameter(required = true)
+    private String kind;
+
+    @Parameter(required = true)
+    private String filenames;
+
+    public String getKind() {
+        return kind;
+    }
+
+    public String getFilenames() {
+        return filenames;
+    }
+
+    public String[] getFilenamesAsArray() {
+        return filenames.split(",\\s*");
```

Comment:
filenames can be null, so should either return an empty string array or null. The caller should throw an exception if this return value then is null. That's the only way to ensure that a mandatory value is provided.